src/XMakeTasks/Copy.cs

```diff
@@ -408,6 +409,20 @@ CopyFileWithState copyFile
                 _copiedFiles = new TaskItem[0];
                 return true;
             }
+            
+            if (_destinationFiles != null)
+            {
+                var sameDestinationFiles = _destinationFiles.GroupBy(x => x).Where(g => g.Count() > 1).Select(y => y.First());
```

Comment:
That output looks pretty good, modulo grammar nitpicks. But I hope we can avoid doing quite so much work to get it. Could you set metadata while populating `_destinationFiles`--maybe something like `Source` or `CopiedFrom`?